src/capi.rs

```diff
@@ -44,8 +52,10 @@ pub unsafe extern "C" fn microvmi_init(
                 .expect("Failed to convert DriverInitParam C struct to Rust equivalent"),
         )
     };
-    let driver = init(&safe_domain_name, optional_driver_type, init_option);
-    Box::into_raw(Box::new(driver)) as *mut c_void
+    match init(&safe_domain_name, optional_driver_type, init_option) {
```

Comment:
Ok, so cbindgen obviously has problems converting complex enums into C structs, but that shouldn't be surprising because there is no real counterpart for those in C. Maybe a union type wrapping multiple structs. However I would not try to replicate rust error handling in C because idiomatic error handling in C just works differently. I would simply return null in error cases and additionally add an out parameter so we can set error values. The error value could be a simply enum or maybe a struct containing an enum and a string description of the error.